Dockerfile-rails

```diff
@@ -1,11 +1,16 @@
 FROM ruby:2.6.2
+
 RUN curl -sL https://deb.nodesource.com/setup_9.x | bash -
 RUN apt-get update -qq && apt-get install -y build-essential libpq-dev postgresql-client nodejs
-ADD Gemfile Gemfile
-ADD Gemfile.lock Gemfile.lock
-ADD package.json package.json
+RUN npm install -g yarn n
+
+# To install the specific version of Node.js
+RUN n 11.14.0
+RUN apt purge -y nodejs && apt autoremove -y
+
+ADD . /app
+WORKDIR /app
+
 RUN bundle install
-RUN npm i
-COPY . /frontend
-WORKDIR /frontend
-RUN npm run dev-build
\ No newline at end of file
+RUN yarn
+RUN yarn dev-build
```

Comment:
To enable docker's layer caching, these lines should be executed before adding whole app files.
It would be better below.
```
WORKDIR /app

ADD Gemfile Gemfile
ADD Gemfile.lock Gemfile.lock
RUN bundle install

ADD package.json package.json
ADD yarn.lock yarn.lock
RUN yarn

ADD . /app
RUN yarn dev-build
```

Reply:
Is it not good enough only to use docker volume based caching?

Reply:
Since docker volume is mounted dynamically when a container of its image is created, `bundle install` will be executed every time whether Gemfile.lock is updated or not.

Reply:
Exactly, thanks.

Reply:
Um, I wonder if this container environment is just for development, not for production deployment, right?
then, it's enough to use just only docker volume-based caching.

Reply:
Yes, I'm planning to use docker-compose only for test/development.